avm/res/web/connection/main.bicep

```diff
@@ -84,24 +143,25 @@ resource connection 'Microsoft.Web/connections@2016-06-01' = {
     displayName: displayName
     customParameterValues: customParameterValues
     api: api
-    parameterValues: parameterValues
-    nonSecretParameterValues: nonSecretParameterValues
+    parameterValues: !empty(parameterValues) ? parameterValues : null
+    nonSecretParameterValues: !empty(nonSecretParameterValues) ? nonSecretParameterValues : null
```

Comment:
Why the empty checks? The parameters are anyways nullable and you're essentially just configuring something like "if the value is not empty/null, set it, otherwise null. Which is pretty much what the nullable parameter claims to begin with. There is one difference though - that is if somebody would set an empty value, your suggested implementation would interpret is as `null`, while the original would pass the empty string to the RP.

Reply:
I was working some more on the parameters and types and this is a leftover that I missed.